File: lib/models/booking.dart
class Booking {
  int? id;
  String customerEmail;
  String serviceName;
  String location;
  String notes;
  String bookingDate;
  String bookingTime;
  int? carId;
  String? carModel;
  String? carPlate;

  // ✅ NEW
  String status;

  Booking({
    this.id,
    required this.customerEmail,
    required this.serviceName,
    required this.location,
    this.notes = '',
    required this.bookingDate,
    required this.bookingTime,
    this.carId,
    this.carModel,
    this.carPlate,
    this.status = 'pending', // default
  });

  Map<String, dynamic> toJson() => {
        'id': id,
        'customer_email': customerEmail,
        'service_name': serviceName,
        'location': location,
        'notes': notes,
        'booking_date': bookingDate,
        'booking_time': bookingTime,
        'car_model': carModel,
        'car_plate': carPlate,

        // ✅ NEW (اختياري إذا بدك تبعتيه)
        'status': status,
      };

  factory Booking.fromJson(Map<String, dynamic> json) => Booking(
        id: json['id'],
        customerEmail: (json['customer_email'] ?? '').toString(),
        serviceName: (json['service_name'] ?? '').toString(),
        location: (json['location'] ?? '').toString(),
        notes: (json['notes'] ?? '').toString(),
        bookingDate: (json['booking_date'] ?? '').toString(),
        bookingTime: (json['booking_time'] ?? '').toString(),
        carId: json['car_id'] == null ? null : int.tryParse(json['car_id'].toString()),
        carModel: json['car_model']?.toString(),
        carPlate: json['car_plate']?.toString(),

        // ✅ NEW
        status: (json['status'] ?? 'pending').toString(),
      );
}

File: carwash_api/booking.php
<?php
header("Access-Control-Allow-Origin: *");
header("Content-Type: application/json; charset=UTF-8");
header("Access-Control-Allow-Methods: POST");
header("Access-Control-Allow-Headers: Content-Type");

// Turn off PHP errors in output
error_reporting(0);
ini_set('display_errors', 0);

include 'config.php';

// Helper function for JSON response
function sendResponse($status, $message, $extra = []) {
    echo json_encode(array_merge([
        "status" => $status,
        "message" => $message
    ], $extra));
    exit;
}

// Car plate validation
function isValidCarPlate($plate) {
    return preg_match('/^[A-Z0-9]{6,8}$/i', $plate);
}

// Read JSON input
$input = file_get_contents("php://input");
$data = json_decode($input);

if (!$data) {
    sendResponse(false, "Invalid JSON input");
}

// Required fields check
$required = ["customer_email", "service_name", "location", "booking_date", "booking_time", "car_model", "car_plate"];
foreach ($required as $field) {
    if (empty($data->$field)) {
        sendResponse(false, "Field '$field' is required");
    }
}

// Car plate validation
if (!isValidCarPlate($data->car_plate)) {
    sendResponse(false, "Invalid car plate number. Must be 6-8 letters/numbers.");
}

try {
    $conn->beginTransaction();

    // 1️⃣ Get user ID
    $stmt = $conn->prepare("SELECT id FROM users WHERE email = ?");
    $stmt->execute([$data->customer_email]);
    $user = $stmt->fetch(PDO::FETCH_ASSOC);

    if (!$user) {
        $conn->rollBack();
        sendResponse(false, "User not found");
    }

    $userId = $user['id'];

    // 2️⃣ Insert car
    $stmt = $conn->prepare("INSERT INTO cars (user_id, car_model, car_plate) VALUES (?, ?, ?)");
    $stmt->execute([$userId, $data->car_model, $data->car_plate]);
    $carId = $conn->lastInsertId();

    // 3️⃣ Find available team (round-robin)
    $teamStmt = $conn->prepare("
        SELECT t.team_id
        FROM teams t
        WHERE t.team_id NOT IN (
            SELECT team_id FROM team_bookings
            WHERE booking_date = ? AND booking_time = ?
        )
        ORDER BY t.team_id ASC
        LIMIT 1
    ");
    $teamStmt->execute([$data->booking_date, $data->booking_time]);
    $team = $teamStmt->fetch(PDO::FETCH_ASSOC);

    if (!$team) {
        $conn->rollBack();
        sendResponse(false, "No available teams at this time");
    }

    $teamId = $team['team_id'];

    // 4️⃣ Insert booking
    $stmt = $conn->prepare("
        INSERT INTO bookings 
        (customer_email, service_name, location, notes, booking_date, booking_time, car_id, team_id)
        VALUES (?, ?, ?, ?, ?, ?, ?, ?)
    ");
    $notes = !empty($data->notes) ? $data->notes : '';
    $stmt->execute([
        $data->customer_email,
        $data->service_name,
        $data->location,
        $notes,
        $data->booking_date,
        $data->booking_time,
        $carId,
        $teamId
    ]);
    $bookingId = $conn->lastInsertId();

    // 5️⃣ Lock team slot
    $stmt = $conn->prepare("
        INSERT INTO team_bookings (team_id, booking_date, booking_time, booking_id)
        VALUES (?, ?, ?, ?)
    ");
    $stmt->execute([$teamId, $data->booking_date, $data->booking_time, $bookingId]);

    $conn->commit();

    sendResponse(true, "Booking created & team assigned", ["team_id" => $teamId]);

} catch (Exception $e) {
    $conn->rollBack();
    sendResponse(false, "Server error: " . $e->getMessage());
}
?>
